optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -941,15 +968,17 @@ public boolean hasMovableEntities(ScoreDirector<Solution_> scoreDirector) {
      * @return {@code >= 0}
      */
     public long getGenuineVariableCount(Solution_ solution) {
-        return extractAllEntitiesStream(solution)
-                .mapToLong(entity -> findEntityDescriptorOrFail(entity.getClass()).getGenuineVariableCount())
-                .sum();
+        LongAdder adder = new LongAdder();
+        visitAllEntities(solution,
+                entity -> adder.add(findEntityDescriptorOrFail(entity.getClass()).getGenuineVariableCount()));
```

Comment:
`visitAllEntities(...)` is more efficient that the previous `extractAllEntitiesStream(...)`.
Where the latter did a lot of string concatenation and then iteration, the former only does the iteration.